packages/financial-templates-lib/src/price-feed/CreatePriceFeed.js

```diff
@@ -107,28 +107,34 @@ async function createPriceFeed(logger, web3, networker, getTime, config) {
       config.priceFeedDecimals, // This defaults to 18 unless supplied by user
       uniswapBlockCache
     );
-  } else if (config.type === "defipulsetvl") {
-    const requiredFields = ["lookback", "minTimeBetweenUpdates", "defipulseApiKey"];
+  } else if (config.type === "defipulse") {
+    const requiredFields = ["lookback", "minTimeBetweenUpdates", "defipulseApiKey", "project"];
+    const VALID_PROJECTS = ["all", "SushiSwap", "Uniswap"];
```

Comment:
For example, add an `assert(VALID_PROJECTS.includes(project), "invalid project name")` to the constructor and then add the requisite tests

Reply:
I'm trying to write a test to check that an error is thrown when an invalid name is passed as a param in DefiPulsePriceFeed.js 

`it("Bad project name", async function() {

    badProjectPriceFeed = new DefiPulsePriceFeed(
      dummyLogger,
      web3,
      defipulseApiKey,
      lookback,
      networker,
      getTime,
      minTimeBetweenUpdates,
      decimals, // Add arbitrary decimal conversion and prove this works.
      "NotAProject"
    );
   
    assert(badProjectPriceFeed instanceof assert.AssertionError);

  });`

The problem is that my test throws an error because DefiPulsePriceFeed() thows an assert error

Reply:
Any tips would be greatly appreciated!

Reply:
theres probably cleverer ways but ive seen this in some other tests.  you could also assert on error.message that it matches expected message
```js
            let errorThrown = false;
            try {
              await liquidator.liquidatePositions();
            } catch (error) {
              errorThrown = true;
            }
            assert.isTrue(errorThrown);
```

Reply:
yeah you could also do probably do something like:

```
(new PriceFeed(...badConfigs)).catch(err => assert(true)).then(assert(false)
```